src/ucp/dt/datatype_iter.inl

```diff
@@ -32,13 +32,19 @@ static UCS_F_ALWAYS_INLINE void
 ucp_datatype_contig_iter_init(ucp_context_h context, void *buffer, size_t length,
                               ucp_datatype_t datatype, ucp_datatype_iter_t *dt_iter)
 {
-    dt_iter->mem_type               = ucp_memory_type_detect(context, buffer,
-                                                             length);
+    ucp_memory_detect(context, buffer, length, &dt_iter->mem_info);
     dt_iter->length                 = length;
     dt_iter->type.contig.buffer     = buffer;
     dt_iter->type.contig.reg.md_map = 0;
 }
 
+static UCS_F_ALWAYS_INLINE void
+ucp_datatype_iter_set_host_mem(ucp_datatype_iter_t *dt_iter)
+{
+    dt_iter->mem_info.type    = UCS_MEMORY_TYPE_HOST;
+    dt_iter->mem_info.sys_dev = UCS_SYS_DEVICE_ID_UNKNOWN;
+}
+
```

Comment:
minor:
maybe it is worth to define `ucp_mem_type_set_host_mem( ucs_memory_info_t *mem_info)` instead?
Then you could use it here and in several other places